File: SeriesGuide/src/main/java/com/battlelancer/seriesguide/ui/TraktCommentsActivity.java
package com.battlelancer.seriesguide.ui;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.ActionBar;
import android.view.MenuItem;
import com.battlelancer.seriesguide.R;

public class TraktCommentsActivity extends BaseActivity {

    public static final int LOADER_ID_COMMENTS = 100;

    /**
     * Display comments of an episode.
     */
    public static Bundle createInitBundleEpisode(String title, int episodeTvdbId) {
        Bundle extras = new Bundle();
        extras.putInt(TraktCommentsFragment.InitBundle.EPISODE_TVDB_ID, episodeTvdbId);
        extras.putString(InitBundle.TITLE, title);
        return extras;
    }

    /**
     * Display comments of a show.
     */
    public static Bundle createInitBundleShow(String title, int showTvdbId) {
        Bundle extras = new Bundle();
        extras.putInt(TraktCommentsFragment.InitBundle.SHOW_TVDB_ID, showTvdbId);
        extras.putString(InitBundle.TITLE, title);
        return extras;
    }

    /**
     * Display comments of a movie.
     */
    public static Bundle createInitBundleMovie(String title, int movieTmdbId) {
        Bundle extras = new Bundle();
        extras.putInt(TraktCommentsFragment.InitBundle.MOVIE_TMDB_ID, movieTmdbId);
        extras.putString(InitBundle.TITLE, title);
        return extras;
    }

    private interface InitBundle {
        String TITLE = "title";
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_singlepane);
        setupActionBar();

        if (savedInstanceState == null) {
            Fragment f = new TraktCommentsFragment();
            f.setArguments(getIntent().getExtras());
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.content_frame, f)
                    .commit();
        }
    }

    @Override
    protected void setupActionBar() {
        super.setupActionBar();
        String commentsTitle = getIntent().getExtras().getString(InitBundle.TITLE);
        setTitle(getString(R.string.comments) + " " + commentsTitle);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setTitle(R.string.comments);
            actionBar.setSubtitle(commentsTitle);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int itemId = item.getItemId();
        if (itemId == android.R.id.home) {
            onBackPressed();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/battlelancer/seriesguide/ui/comments/TraktCommentsActivity.kt
package com.battlelancer.seriesguide.ui.comments

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import com.battlelancer.seriesguide.R
import com.battlelancer.seriesguide.ui.BaseActivity
import timber.log.Timber

/**
 * Hosts a [TraktCommentsFragment].
 */
class TraktCommentsActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_singlepane)

        if (intent.extras == null) {
            finish()
            Timber.e("Finishing, missing required extras.")
            return
        }

        setupActionBar()

        if (savedInstanceState == null) {
            val f = TraktCommentsFragment().apply {
                arguments = intent.extras
            }
            supportFragmentManager.beginTransaction()
                .add(R.id.content_frame, f)
                .commit()
        }
    }

    override fun setupActionBar() {
        super.setupActionBar()
        val commentsTitle: String? = intent.getStringExtra(EXTRA_TITLE)
        title = "${getString(R.string.comments)} $commentsTitle"
        supportActionBar?.apply {
            setHomeAsUpIndicator(R.drawable.ic_clear_24dp)
            setDisplayHomeAsUpEnabled(true)
            setTitle(R.string.comments)
            subtitle = commentsTitle
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                onBackPressed()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    companion object {

        const val LOADER_ID_COMMENTS = 100
        private const val EXTRA_TITLE = "title"

        /**
         * Display comments of an episode.
         */
        @JvmStatic
        fun intentEpisode(context: Context, title: String?, episodeId: Long): Intent {
            return Intent(context, TraktCommentsActivity::class.java)
                .putExtra(TraktCommentsFragment.InitBundle.EPISODE_ID, episodeId)
                .putExtra(EXTRA_TITLE, title)
        }

        /**
         * Display comments of a show.
         */
        fun intentShow(context: Context, title: String?, showId: Long): Intent {
            return Intent(context, TraktCommentsActivity::class.java)
                .putExtra(TraktCommentsFragment.InitBundle.SHOW_ID, showId)
                .putExtra(EXTRA_TITLE, title)
        }

        /**
         * Display comments of a movie.
         */
        fun intentMovie(context: Context, title: String?, movieTmdbId: Int): Intent {
            return Intent(context, TraktCommentsActivity::class.java)
                .putExtra(TraktCommentsFragment.InitBundle.MOVIE_TMDB_ID, movieTmdbId)
                .putExtra(EXTRA_TITLE, title)
        }
    }
}
